### PR TITLE
Add forms for system SMS configs and sharing logic

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes GH-432

**Release Note**

```release-note
Add forms for system SMS configs and sharing logic
```

### CODE REVIEW

File: pkg/database/sms_config.go [hunk 1]
```diff
@@ -71,6 +85,20 @@ func (db *Database) SaveSMSConfig(s *SMSConfig) error {
 			return nil
 		}
 
+		if s.IsSystem {
+			// We're about to delete the system SMS config, revoke everyone's
+			// permissions to use it. You would think there'd be a way to do this with
+			// gorm natively. You'd even find an example in the documentation that led
+			// you to believe that gorm does this.
+			//
+			// Narrator: gorm does not do this.
```
Comment (whaught) [inline pkg/database/sms_config.go:94]: This surprises me

**Review by whaught — COMMENTED** [review]
/lgtm

File: pkg/database/sms_config.go (hunk 1 above)
Comment (sethvargo) [reply pkg/database/sms_config.go:94]: Yea, if you actually print out the SQL for the example in gorm's docs, it's `SELECT * FROM realms WHERE id = '' UPDATE ...` which isn't even valid SQL

Comment (whaught) [reply pkg/database/sms_config.go:94]: ORMs are probably expecting row-wise object updates rather than a blind update of a whole column

**Pull request merged**